meilisearch-http/src/backup.rs

```diff
@@ -0,0 +1,369 @@
+
+    Ok(metadata)
+}
+
+fn settings_to_path(settings: &Settings, folder_path: &Path) -> Result<(), Error> {
+    let path = folder_path.join("settings.json");
+    let file = File::create(path)?;
+
+    serde_json::to_writer(file, settings)?;
+
+    Ok(())
+}
+
+fn import_index_v1(
+    data: &Data,
+    backup_path: &Path,
+    index_uid: &str,
+    document_batch_size: usize,
+    write_txn: &mut RwTxn<MainT>,
+) -> Result<(), Error> {
+    let index = data
+        .db
+        .open_index(index_uid)
+        .ok_or(Error::index_not_found(index_uid))?;
+
+    let index_path = &backup_path.join(index_uid);
+
+    let settings = settings_from_path(&index_path)?;
+    let settings = settings.to_update().or_else(|_e| Err(Error::backup_failed()))?;
+    apply_settings_update(write_txn, &index, settings)?;
+
+    let documents = {
+        let file = File::open(&index_path.join("documents.jsonl"))?;
+        let reader = std::io::BufReader::new(file);
+        let deserializer = serde_json::Deserializer::from_reader(reader);
+        deserializer.into_iter::<serde_json::Value>()
+    };
+
+    let mut values = Vec::with_capacity(document_batch_size);
+    for document in documents {
+        let vec = serde_json::to_vec(&document?)?;
+        let document = serde_json::from_slice(&vec)?;
+        values.push(document);
+        if values.len() == document_batch_size { 
+            apply_documents_addition(write_txn, &index, values.clone())?;
+            values.clear();
+        }
+    }
+
+    // apply last documents
+    if values.len() > 0 { 
+        apply_documents_addition(write_txn, &index, values)?;
+    }
+
+    Ok(())
+}
+
+pub fn import_backup(
+    data: &Data,
+    backup_path: &Path,
+    document_batch_size: usize,
+) -> Result<(), Error> {
+    let tmp_dir = TempDir::new()?;
+    let tmp_dir_path = tmp_dir.path();
+
+    compression::from_tar_gz(backup_path, tmp_dir_path)?;
+
+    let metadata = BackupMetadata::from_path(&tmp_dir_path)?;
+
+    let import_index = match metadata.backup_version {
+        BackupVersion::V1 => import_index_v1,
+    };
+
+    // clear database
+    for index_uid in data.db.indexes_uids() {
+        data.db.delete_index(index_uid)?;
+    }
+
+    // create new indexes
+    for index in metadata.indexes.iter() {
+            index::create_index_sync(&data.db, index.uid.clone(), index.name.clone(), index.primary_key.clone())?;
+    }
+
+    data.db.main_write::<_, _, Error>(|mut writer| {
+        for index in metadata.indexes {
+            import_index(&data, tmp_dir_path, &index.uid, document_batch_size, &mut writer)?;
+        }
+        Ok(())
+    })?;
+
+    Ok(())
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum BackupStatus {
+    Processing,
+    Done,
+    BackupProcessFailed,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct BackupResume {
```

Comment:
BackupInfo? BackupSummary? BackupState? ('Resume' may be confusing, sounds like some task was paused and should be continued...)